include/deal.II/lac/solver_qmrs.h

```diff
@@ -74,86 +94,101 @@ class SolverQMRS : public Solver<VectorType>
   /**
    * Standardized data struct to pipe additional data to the solver.
    *
-   * There are two possibilities to compute the residual: one is an estimate
-   * using the computed value @p tau. The other is exact computation using
-   * another matrix vector multiplication.
+   * The user is able to switch between right and left preconditioning, that means
+   * solving the systems <i>P<sup>-1</sup>A</i> and <i>AP<sup>-1</sup></i> respectively,
+   * using the corresponding parameter. Note, that left preconditioning means to
+   * employ the preconditioned (BiCG-)residual and otherwise the unpreconditioned one.
+   * The default is the application from the right side.
+   *
+   * The solver_tolerance threshold is used to define the said bound below which the residual
+   * is computed exactly. See class documentation for more information. The default value is 1e-9,
+   * that is the default solving precision multiplied by ten.
    *
-   * QMRS, is susceptible to breakdowns, so we need a parameter telling us,
+   * SQMR is susceptible to breakdowns (divisions by zero), so we need a parameter telling us,
    * which numbers are considered zero. The proper breakdown criterion is very
-   * unclear, so experiments may be necessary here.
+   * unclear, so experiments may be necessary here. It is even possible to achieve convergence
+   * despite of dividing through small numbers. There are even cases, in which it is advantageous to
+   * accept such divisions because the cheap iteration cost makes the algorithm the fastest of all
+   * available indefinit solvers. Nonetheless, the default breakdown threshold value is 1e-16.
    */
   struct AdditionalData
   {
     /**
      * Constructor.
      *
-     * The default is no exact residual computation and breakdown parameter
+     * The default is right preconditioning and breakdown parameter
      * 1e-16.
      */
     explicit
-    AdditionalData(const bool   exact_residual = false,
-                   const double breakdown      = 1.e-16) :
-      exact_residual(exact_residual),
-      breakdown(breakdown)
-    {}
+    AdditionalData (bool left_preconditioning = false,
+                    double solver_tolerance = 1.e-9,
+                    bool breakdown_testing = true,
```

Comment:
This is an incompatible change from the old list of arguments to the new one. That's ok, but it should be mentioned in a changelog entry in `doc/news/changes/incompatible`.

Also, since all of these arguments are input arguments, can you mark them as `const`?

Reply:
So small changelog textfile written and every input argument `const` from now on.